src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+            // Refer http://msdn.microsoft.com/en-us/library/b05d59ty.aspx
+            Assert.ThrowsAny<Exception>(() => pi.GetValue(null, new Object[] { "1", "2" }));
+        }
+
+
+
+        [Theory]
+        [InlineData("NoSetterProp", 100, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "hello", "SOME  string" })]
+        [InlineData("Item", 100, new Object[] { 99, 2, new string[] { "hello" }, "SOME  string" })]
+        public static void TestSet_ThrowsArgumentException(string propertyName, Object setValue, Object[] index)
+        {
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                pi.SetValue(obj, setValue, index);
+            });
+        }
+        
+        //Try to set instance Property with null object 
+        [Theory]
+        [InlineData("HasSetterProp", null, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "SOME string" })]
+        public static void TestSetNull_ThrowsException(string propertyName, Object setValue, Object[] index)
+        {
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // Generic Exception is thrown in WP8 instead of TargetException
+            Assert.ThrowsAny<Exception>(() => pi.SetValue(null, null, null));
+        }
+
+        //Try to set instance Property with wrong type
+        [Fact]
+        public static void TestSet_ThrowsException()
+        {
+            string propertyName = "HasSetterProp";
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // Generic Exception is thrown in WP8 instead of TargetException
+            Assert.ThrowsAny<Exception>(() => pi.SetValue(obj, "foo", null));
+        }
+        
+
+        //Try to set indexer Property , Incorrect number of parameters
+        [Fact]
+        public static void TestSet_ThrowsTargetParameterCountException()
+        {
+            string propertyName = "Item";
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<TargetParameterCountException>(() =>
+            {
+                pi.SetValue(obj, "pw", new Object[] { 99, 2, new string[] { "SOME string" } });  // Incorrect number of parameters
+            });
+        }
+       
+
+
+        //Gets PropertyInfo object from a Type
+        public static PropertyInfo GetProperty(Type t, string property)
```

Comment:
Nit: for helpers like this that are only used by this class, they can be made private instead of public.  While this is "just" a test and visibility doesn't really matter from a consumption perspective, making this public IMO makes it easier for someone reading through the tests to understand the intention, and to understand that changes to this function can only impact other code in this class and not external to it.